conda/core/solve.py

```diff
@@ -1324,28 +1345,65 @@ def _configure_solver_for_remove(self, state):
         state["solver"] = solver
         return solver
 
-    def _collect_specs_to_add(self, state, update_modifier=NULL):
+    def _collect_specs_to_add(self,
+            state,
+            update_modifier=NULL,
+            deps_modifier=NULL,
+            ignore_pinned=NULL,
+            force_remove=NULL,
+            force_reinstall=NULL):
         """
         Reimplement the logic found in super()._add_specs(), but simplified.
         """
+        from mamba import mamba_api as api
+        task_and_specs = defaultdict(set)
         # A. Explicitly requested packages via CLI (either update or install)
-        specs_to_add = list(self.specs_to_add)
+        task = (api.SOLVER_UPDATE if self._command == "update" else api.SOLVER_INSTALL)
```

Comment:
No need for extra parentheses.
```suggestion
        task = api.SOLVER_UPDATE if self._command == "update" else api.SOLVER_INSTALL
```